### PR TITLE
Namespaced installation with ProjectRequests and limited roles

### PR DESCRIPTION
**Pull request by matskiv**
JIRAs:
https://issues.redhat.com/browse/INTLY-4238
https://issues.redhat.com/browse/INTLY-4239
https://issues.redhat.com/browse/INTLY-4956
https://issues.redhat.com/browse/INTLY-4909

Update: ready for review and verification now.
Ping @philbrookes @jjaferson 

One possibly controversial change - I moved CRD watch creation into the end of reconcile process.
Motivation for that - the watches can't be created before the namespace is there. And by moving it there, we can remove CRD detector.

For proper verification this needs to be run with the new roles, e.g. via `make code/run/service_account`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -183,7 +180,7 @@ cluster/cleanup/crds:
 deploy/integreatly-installation-cr.yml: export SELF_SIGNED_CERTS := true
 deploy/integreatly-installation-cr.yml: export INSTALLATION_NAME := integreatly-operator
 deploy/integreatly-installation-cr.yml: export INSTALLATION_TYPE := managed
-deploy/integreatly-installation-cr.yml: export INSTALLATION_PREFIX := rhmi
+deploy/integreatly-installation-cr.yml: export INSTALLATION_PREFIX := integreatly-
```
Comment (philbrookes) [inline Makefile]: Should we stick to one or the other here? https://github.com/integr8ly/integreatly-operator/pull/323/files#diff-a35ad113285fb0512c183bad2b0aeaebR7

File: test/e2e/integreatly_test.go [hunk 2]
```diff
@@ -117,34 +102,70 @@ func integreatlyMonitoringTest(t *testing.T, f *framework.Framework, ctx *framew
...
-			if promApiCallOutput.Data.Alerts[a].State == "firing" {
-				firingalerts = append(firingalerts, promApiCallOutput.Data.Alerts[a].Labels.Alertname)
+			if alert.State == "firing" {
+				firingalerts = append(firingalerts, string(alert.Labels["alertname"]))
 			}
-			if promApiCallOutput.Data.Alerts[a].State == "pending" {
-				pendingalerts = append(pendingalerts, promApiCallOutput.Data.Alerts[a].Labels.Alertname)
+			if alert.State == "pending" {
+				pendingalerts = append(pendingalerts, string(alert.Labels["alertname"]))
+			}
```
Comment (davidffrench) [inline test/e2e/integreatly_test.go]: @matskiv Will pending alerts return in the `/alerts` endpoint?

Comment (matskiv) [reply test/e2e/integreatly_test.go]: @davidffrench I don't know. I was curious about this as well, but I decided to leave it as is and get us unblocked ASAP. 
And I deleted my installation, so I can't check it now :(

Comment (davidffrench) [reply test/e2e/integreatly_test.go]: thanks @matskiv . Would be good to know this.

**Comment by matskiv** [comment]
In the "Fix uninstallation of ProjectRequest based install" I added a check into some finalizer functions.
It's only for the finalizer functions that check for some resources in a product namespace. The check will verify that a namespace still exists. This will prevent "forbidden" errors when trying to list/get resources of a removed namespace.

**Pull request merged**